### PR TITLE
[RDF] Use CMS OpenData from 2012 for NanoAOD example producing dimuon…

### PR DESCRIPTION
**Pull request by stwunsch**
… spectrum and adjust final plot

The files are taken form publiceos!

The final plot looks like this:

x

**Edit:**

I've restuctured the tutorial a little and simplified the cuts. Now, it's very clear without index magic and much more condense! The simplification comes from requiring *exactly* two muons for each events. We loose a little statistics but the analysis is now easier to understand. I leave the complicated stuff to the upcoming Higgs analysis tutorial ;) The plot looks now like this:

x

And I've added a cut-flow report! ->

```
Events with exactly two muons: pass=33370298   all=66128870   --   50.463 %
Muons with opposite charge: pass=25794885   all=33370298   --   77.299 %
```

Runtime depends heavily on the xrootd connection. From my experience about 1 to 3 minutes with files read remotely and about 20s with files read from local disk.

### CODE REVIEW

**Comment by stwunsch** [comment]
Only problem so far:
The runtime is heavily dependent on pulling the data from the source. eospublic.cern.ch+xrootd seems to be even slower than root.cern.ch+davix. A full run on my laptop takes about 15min to 30min (why this fluctuates so much?) compared to about 40s having the files locally.

File: tutorials/dataframe/df102_NanoAODDimuonAnalysis.C [hunk 1]
```diff
@@ -75,39 +77,38 @@ void df102_NanoAODDimuonAnalysis()
...
 
    // Produce histogram of di-muon mass spectrum
-   auto h = df_mass.Histo1D({"Dimuon_mass", "Dimuon_mass", 20000, 0.25, 300}, "Dimuon_mass")
+   auto h = df_mass.Histo1D({"Dimuon_mass", "Dimuon_mass", 30000, 0.25, 300}, "Dimuon_mass")
                    .GetValue();
 
    // Make plot
    gStyle->SetOptStat(0);
    gStyle->SetTextFont(42);
-   auto c = new TCanvas("c", "c", 800, 600);
```
Comment (dpiparo) [inline tutorials/dataframe/df102_NanoAODDimuonAnalysis.C]: It is not obvious but the leak is on purpose: we want the canvas and histo and the rest survive the scope in order for our doxy filter to inline the plot in the documentation page.

Comment (stwunsch) [reply tutorials/dataframe/df102_NanoAODDimuonAnalysis.C]: Done!

File: tutorials/dataframe/df102_NanoAODDimuonAnalysis.C [hunk 2]
```diff
@@ -30,84 +30,58 @@ void df102_NanoAODDimuonAnalysis()
...
-   ROOT::RDataFrame df("Events", "http://root.cern.ch/files/NanoAOD_DoubleMuon_CMS2011OpenData.root");
+   // Create dataframe from NanoAOD files
+   ROOT::RDataFrame df("Events",
+                      {"root://eospublic.cern.ch//eos/root-eos/cms_opendata_2012_nanoaod/Run2012B_DoubleMuParked.root",
+                       "root://eospublic.cern.ch//eos/root-eos/cms_opendata_2012_nanoaod/Run2012C_DoubleMuParked.root"});
 
-   // Select events with more than two muons
-   auto df_filtered = df.Filter("nMuon>=2", "More than two muons");
+   // Select events with exactly two muons and opposite charge
+   auto df_2mu = df.Filter("nMuon == 2", "Events with exactly two muons");
```
Comment (dpiparo) [inline tutorials/dataframe/df102_NanoAODDimuonAnalysis.C]: Comment about this simplification and mention that we are loosing some events.

File: tutorials/dataframe/df102_NanoAODDimuonAnalysis.C [hunk 3]
```diff
@@ -30,84 +30,58 @@ void df102_NanoAODDimuonAnalysis()
...
-   label.DrawLatex(0.170, 0.200, "#bf{L_{int} = 2.4 fb^{-1}}");
-   label.SetTextSize(0.032);
-   label.DrawLatex(0.10, 0.920, "Run2011A Double Muon Dataset (DOI: 10.7483/OPENDATA.CMS.RZ34.QR6N)");
+   label.DrawLatex(0.415, 0.670, "#psi'");
+   label.DrawLatex(0.485, 0.700, "Y(1,2,3S)");
+   label.DrawLatex(0.755, 0.680, "Z");
+   label.SetTextSize(0.040); label.DrawLatex(0.100, 0.920, "#bf{CMS Open Data}");
+   label.SetTextSize(0.030); label.DrawLatex(0.630, 0.920, "#sqrt{s} = 8 TeV, L_{int} = 11.6 fb^{-1}");
+
+   c->SaveAs("dimuon_spectrum.pdf");
```
Comment (dpiparo) [inline tutorials/dataframe/df102_NanoAODDimuonAnalysis.C]: Add a comment about the prescales in order not to have people commenting about the bump at 30 GeV :)

Comment (dpiparo) [reply tutorials/dataframe/df102_NanoAODDimuonAnalysis.C]: just for the beginners. Let's call it "Trigger effect"

**Pull request merged**